frontend/src/features/settings/hooks/creditAccount.ts

```diff
@@ -0,0 +1,254 @@
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import {
+  addCreditAccountUser,
+  createCreditAccount,
+  getCreditAccount,
+  removeCreditAccountUsers,
+  getCreditAccountUsers,
+  updateCreditAccountStatus,
+  getCompanyCreditAccount,
+} from "../apiManager/creditAccount";
+import { getCompanyDataBySearch } from "../../idir/search/api/idirSearch";
+import { useNavigate } from "react-router-dom";
+import { ERROR_ROUTES } from "../../../routes/constants";
+import { SnackBarContext } from "../../../App";
+import { useContext } from "react";
+import {
+  CreditAccountStatusType,
+  UPDATE_STATUS_ACTIONS,
+  UpdateStatusData,
+} from "../types/creditAccount";
+import { getCompanyIdFromSession } from "../../../common/apiManager/httpRequestHandler";
+import { SnackbarAlertType } from "../../../common/components/snackbar/CustomSnackBar";
+import { CompanyProfile } from "../../manageProfile/types/manageProfile";
+
+/**
+ * Hook to fetch the company credit account details for the active user.
+ * @returns Query result of the company credit account details
+ */
+export const useGetCreditAccountQuery = () => {
+  return useQuery({
+    queryKey: ["credit-account", { companyId: getCompanyIdFromSession() }],
```

Comment:
Although it's possible to use getCompanyIdFromSession here inside this hook, I still recommend call it inside the parent component and passing its result into this hook as a param, ie. ```export const useGetCreditAccountQuery = (companyId: string | number) => ...```